### DISCUSSION TITLE
including-linked-files-from-outside-the-project-directory-in-asp-net-core/

### DISCUSSION THREAD

**Discussion by andrewlock** [andrewlock/blog-comments#246] [Announcements]
# including-linked-files-from-outside-the-project-directory-in-asp-net-core/

In this post I show how to include files in a project that are outside the project directory

https://andrewlock.net/including-linked-files-from-outside-the-project-directory-in-asp-net-core/

[a comment by an automated account is not shown]

**Reply by andrewlock** [reply]
<em>[Sock](https://disqus.com/by/andrewlockdotnet/) (@andrewlock) commented [on Disqus](https://disqus.com/home/forum/andrewlock/) at March 07 2018, 02:20, in reply to [Paul Van Bladel](https://disqus.com/by/paulvanbladel/)'s [comment](https://github.com/andrewlock/blog-comments/discussions/246#discussioncomment-6045711)</em>

---

<p>Could you give more detail? Do you mean when developing on Linux/Docker the linking doesn't work?</p>

**Reply by ejcoutinho** [reply]
I came to this page with same problem. I have a shared Lib folder, just with code, and created a link as you sugest, and then I have the error:
 #13 [build 7/7] RUN dotnet build "./Test01.csproj" -c Release -o /app/build
#13 0.592 MSBuild version 17.3.2+561848881 for .NET
#13 1.432   Determining projects to restore...
#13 1.912   All projects are up-to-date for restore.
#13 2.949 CSC : error CS2001: Source file '/src/../../../src/Lib/Const/UDB.cs' could not be found. [/src/Test01.csproj]
It compiles and runs on debuguer, but when I try to publish it, have this error.

I'm trying multiple aproachs to keep the shared Lib folder, but seems all fail in some kind of problem when publish - it doesn't follow the links. I tried include projects; linked folders, and now linked files without sucess.

[a comment by an automated account is not shown]
